### PR TITLE
Build/create season module

### PR DESCRIPTION
**Pull request by Makarona10**

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -6,17 +6,22 @@ import { AxiosModule } from './services/axios/axios.module';
...
+import { SeasonModule } from './season/season.module';
 
 @Module({
   imports: [
     ConfigModule.forRoot({
       isGlobal: true,
     }),
-    MongooseModule.forRoot(process.env.MONGODB_URL),
+    // MongooseModule.forRoot(process.env.MONGODB_URL),
+    MongooseModule.forRoot("mongodb://localhost:27017"),
```
Comment (AMRRAGHEB118) [inline src/app.module.ts]: Why is line 17 commented out, env should be used in this case.

File: src/player/player.controller.ts [hunk 2]
```diff
@@ -18,8 +27,61 @@ export class PlayerController {
...
+            `Found player ${id} successfully`,
+            "player/:id", "GET", 200, LoggerModule.PLAYER
+          );
+        return response.status(200)
+          .send
+          (
+            {
+              "message": "Player found successfully.",
+              "status_code": 200,
+              "data": [result.data]
```
Comment (AMRRAGHEB118) [inline src/player/player.controller.ts:48]: Use dict, it's better than list in this case in my opinion.

File: src/player/player.service.ts [hunk 3]
```diff
@@ -1,36 +1,49 @@
...
+        if (result.status_code == 200) {
+          player = await this.playerModel
+            .findOne({ id })
+            .populate('statistics')
+            .exec();
+          result.data = player;
+        }
+        return result;
+      }
+      return {
```
Comment (AMRRAGHEB118) [inline src/player/player.service.ts:36]: What's error here ?

File: src/season/schema/season.schema.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  id: number;
+
+  @Prop({required: true})
+  name: string;
+
+  @Prop({required: true})
+  leagueId: number;
+
+  @Prop({required: true})
+  teamsIds: number[];
```
Comment (AMRRAGHEB118) [inline src/season/schema/season.schema.ts:16]: Here, the property type is not a number, but an ObjectId.
[ObjectId reference](https://www.mongodb.com/docs/manual/reference/method/ObjectId/)

File: src/season/season.controller.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Controller, Get } from '@nestjs/common';
+import { SeasonService } from './season.service';
+
+@Controller('season')
+export class SeasonController {
+  constructor( private seasonService: SeasonService) {}
+
+  @Get()
+  async getSeasons() {
```
Comment (AMRRAGHEB118) [inline src/season/season.controller.ts:9]: We need the league ID, right ?

File: src/services/dataImport/data.import.service.ts [hunk 6]
```diff
@@ -1,75 +1,192 @@
...
 import { DataMapService } from '../datamap/data-map.service';
-import { Team } from 'src/team/schema/team.schema';
+import { LoggerService } from '../logger/logger.service';
+import { LoggerModule } from '../logger/logger.schema';
 
 
+type Response = {
+  "err": string,
+  "status_code": number,
+  "data": any,
```
Comment (AMRRAGHEB118) [inline src/services/dataImport/data.import.service.ts:12]: Same comment as above.

File: src/services/dataImport/data.import.service.ts [hunk 7]
```diff
@@ -1,75 +1,192 @@
...
+import { Injectable } from '@nestjs/common';
 import { AxiosService } from '../axios/axios.service';
 import { ConfigService } from '@nestjs/config';
 import { DataMapService } from '../datamap/data-map.service';
-import { Team } from 'src/team/schema/team.schema';
+import { LoggerService } from '../logger/logger.service';
+import { LoggerModule } from '../logger/logger.schema';
 
 
+type Response = {
```
Comment (AMRRAGHEB118) [inline src/services/dataImport/data.import.service.ts:9]: It might be better to move it to its own file for global use.

**Review by AMRRAGHEB118 — CHANGES_REQUESTED** [review]

File: src/season/schema/season.schema.ts (hunk 4 above)
Comment (Makarona10) [reply src/season/schema/season.schema.ts:16]: It's the team id of third party that we filter teams based on it before saving to our database.

File: src/season/season.controller.ts (hunk 5 above)
Comment (Makarona10) [reply src/season/season.controller.ts:9]: No, This API just retrieves all seasons (3 seasons) with their league id

**Review by AMRRAGHEB118 — APPROVED** [review]

**Pull request merged**